src/languages/yaml.js

```diff
@@ -102,7 +102,7 @@ export default function(hljs) {
       // Indentation of subsequent lines must be the same to
       // be considered part of the block
       className: 'string',
-      begin: '[\\|>]([0-9]?[+-])?[ ]*\\n( *)[\\S ]+\\n(\\2[\\S ]+\\n?)*'
+      begin: '[\\|>]([0-9]?[+-])?[ ]*\\n( *)[^\\n]+\\n(\\2[^\\n]+\\n?)*'
```

Comment:
Speed optimization... I think this is accomplishing the same thing...

Reply:
`\S` is anything _but_ whitespace. So `[\S ]` to me sounds like "anything but whitespace but also allow a single space character". I don't think tabs or spaces aren't covered by `\n`?

Reply:
But we're using "not \n"... The question to ask here is what was the original intent... aren't we trying to handle code like:

```
key: |
  multi-string
  value
```

Ie, some number of spaces (matching, hence the `\2`) then content (up until a new line)...  I believe `[^\n]` matches the intention, yes?

Reply:
The full expression is `[\S ]+`... not space or space, repeated at least once... and this naturally excludes new lines... so I (I believe) wrote an equivalent and faster version.

Reply:
Taking this snippet out of context, I got the following screenshots:

![image](https://user-images.githubusercontent.com/1246453/97924310-4dfe6080-1d14-11eb-86e3-807271536916.png)
![image](https://user-images.githubusercontent.com/1246453/97924332-56569b80-1d14-11eb-868c-0a9f4780021e.png)
![image](https://user-images.githubusercontent.com/1246453/97924350-5ce51300-1d14-11eb-8350-e3828db7fe6b.png)

`\S` is short for [`[^ \f\n\r\t\v\u00a0\u1680\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Character_Classes)

At the very least, shouldn't `\t` be inside alongside `\n` too?

Reply:
That's clear enough, but I think the prior behavior is WRONG... (unless you want to school me on a finer detail of YAML here - which is possible since YAML is insane)... the intention AFAIK is to go all the way to the end of the line... I'm not sure tabs should break a string sequence... I would expect them to simply be registered as part of the content.

If that is *incorrect* due to some insanity of YAML then I believe you would be right in that we should add `\t`.

Reply:
Ahhh I see what this regex is supposed to do now. I was just comparing the semantics of diffs between regexes. You are correct, there should not be a `\t` in here and previous behavior was wrong 👍